openmc/data/fission_energy.py

```diff
@@ -0,0 +1,627 @@
+        ENDF-102.  Also, some evaluations will give zero uncertainty.  Use with
+        caution.
+
+    """
+    ident = identify_nuclide(filename)
+
+    if not ident['LFI']:
+        # This nuclide isn't fissionable.
+        return None
+
+    # Extract the MF=1, MT=458 section.
+    lines = []
+    with open(filename, 'r') as fh:
+        line = fh.readline()
+        while line != '':
+            if line[70:75] == ' 1458':
+                lines.append(line)
+            line = fh.readline()
+
+    if len(lines) == 0:
+        # No 458 data here.
+        return None
+
+    # Read the number of coefficients in this LIST record.
+    NPL = read_CONT_line(lines[1])[4]
+
+    # Parse the ENDF LIST into an array.
+    data = []
+    for i in range(NPL):
+        row, column = divmod(i, 6)
+        data.append(read_float(lines[2 + row][11*column:11*(column+1)]))
+
+    # Declare the coefficient names and the order they are given in.  The LIST
+    # contains a value followed immediately by an uncertainty for each of these
+    # components, times the polynomial order + 1.
+    labels = ('EFR', 'ENP', 'END', 'EGP', 'EGD', 'EB', 'ENU', 'ER', 'ET')
+
+    # Associate each set of values and uncertainties with its label.
+    value = dict()
+    uncertainty = dict()
+    for i in range(len(labels)):
+        value[labels[i]] = data[2*i::18]
+        uncertainty[labels[i]] = data[2*i + 1::18]
+
+    # In ENDF/B-7.1, data for 2nd-order coefficients were mistakenly not
+    # converted from MeV to eV.  Check for this error and fix it if present.
+    n_coeffs = len(value['EFR'])
+    if n_coeffs == 3:  # Only check 2nd-order data.
+        # Check each energy component for the error.  If a 1 MeV neutron
+        # causes a change of more than 100 MeV, we know something is wrong.
+        error_present = False
+        for coeffs in value.values():
+            second_order = coeffs[2]
+            if abs(second_order) * 1e12 > 1e8:
+                error_present = True
+                break
+
+        # If we found the error, reduce all 2nd-order coeffs by 10**6.
+        if error_present:
+            for coeffs in value.values(): coeffs[2] *= 1e-6
+            for coeffs in uncertainty.values(): coeffs[2] *= 1e-6
+
+        # Perform the sanity check again... just in case.
+        for coeffs in value.values():
+            second_order = coeffs[2]
+            if abs(second_order) * 1e12 > 1e8:
+                raise ValueError("Encountered a ludicrously large second-"
+                                 "order polynomial coefficient.")
+
+    # Convert eV to MeV.
+    for coeffs in value.values():
+        for i in range(len(coeffs)):
+            coeffs[i] *= 10**(-6 + 6*i)
+    for coeffs in uncertainty.values():
+        for i in range(len(coeffs)):
+            coeffs[i] *= 10**(-6 + 6*i)
+
+    return value, uncertainty
+
+
+def write_compact_458_library(endf_files, output_name=None, comment=None,
```

Comment:
This function doesn't use `FissionEnergyRelease` data or methods so I don't think it should be a method.